manifests/server.pp

```diff
@@ -43,6 +43,7 @@
   Boolean $purge_config_dir                                                     = $prometheus::purge_config_dir,
   Boolean $manage_user                                                          = $prometheus::manage_user,
   Optional[Variant[Stdlib::HTTPurl, Stdlib::Unixpath, String[1]]] $external_url = $prometheus::external_url,
+  Optional[Array[Hash[String, String]]] $collect_scrape_jobs                    = [],
```

Comment:
since the default is an empty array, you don't need to make it optional. Also please enforce that the string is at least 1 char long:
```suggestion
  Array[Hash[String[1], String[1]]] $collect_scrape_jobs                        = [],
```